velox/functions/prestosql/TransformKeys.cpp

```diff
@@ -80,7 +80,8 @@ class TransformKeysFunction : public exec::VectorFunction {
           elementToTopLevelRows,
           &transformedKeys);
 
-      if (transformedKeys->mayHaveNulls()) {
+      if (transformedKeys->mayHaveNulls() ||
+          transformedKeys->mayHaveNullsRecursive()) {
```

Comment:
is this an unrelated bug fix?
Also, transformedKeys->mayHaveNullsRecursive() should cover transformedKeys->mayHaveNulls() as well so we only need mayHaveNullsRecursive()

Reply:
This fix was uncovered by the change. Without it a test fails.

I have copied this code from Map.cpp 

```
      // Verify there are no null keys.
      auto keysElements = keysArray->elements();
      if (keysElements->mayHaveNulls() ||
          keysElements->mayHaveNullsRecursive()) {
```

velox/functions/prestosql/MapFromEntries.cpp has similar:

```
    // Validate all map entries and map keys are not null.
    if (decodedRowVector->mayHaveNulls() || keyVector->mayHaveNulls() ||
        keyVector->mayHaveNullsRecursive()) {
```

Maybe we can look into simplifying all these in a follow up.

Reply:
Thanks for the clarification, can you please add a short line in the commit message talking about it as its not obvious how its related to the change? Thanks!

Reply:
Added 

```
This change uncovered a bug in tranform_keys Presto lambda function. The check
for keys containing nulls was incorrectly skipped if there were no null keys. This change includes a fix.
```